### PR TITLE
add check for expected memory layout in numpy

### PR DESCRIPTION
**Pull request by Andrey1994**
Signed-off-by: Andrey Parfenov <a1994ndrey@gmail.com>

https://github.com/brainflow-dev/brainflow/issues/248

### CODE REVIEW

File: python-package/brainflow/utils.py [hunk 1]
```diff
@@ -5,8 +5,6 @@
 
 
 def check_memory_layout_row_major(data):
-    if not data:
```
Comment (xloem) [inline python-package/brainflow/utils.py]: i notice you removed this to fix ci.  another approach might be something like `if data is None or len(data) == 0:`.  i'm not as familiar with python as c++ to know what is normal.

File: tests/python/hardware_test.py [hunk 2]
```diff
@@ -1,174 +0,0 @@
-import argparse
```
Comment (xloem) [inline tests/python/hardware_test.py:1]: did you mean to delete this file?  it's not mentioned in your change summaries.

File: python-package/brainflow/data_filter.py [hunk 3]
```diff
@@ -383,6 +384,7 @@ def perform_lowpass(cls, data: NDArray[Float64], sampling_rate: int, cutoff: flo
         :param ripple: ripple value for Chebyshev filter
         :type ripple: float
         """
+        check_memory_layout_row_major(data)
```
Comment (xloem) [inline python-package/brainflow/data_filter.py:387]: note that this is the same conceptual kind of check as https://github.com/brainflow-dev/brainflow/pull/253/files#diff-d5c3bcf84e585e262092315dad35c121847283e2084c58a0281297dde4cfcbf0R392 and applies to all the same places, so the code could be merged

File: tests/python/hardware_test.py (hunk 2 above)
Comment (Andrey1994) [reply tests/python/hardware_test.py:1]: yep, its intentional

File: python-package/brainflow/utils.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import numpy as np
+
+from brainflow.board_shim import BrainFlowError
+from brainflow.exit_codes import BrainflowExitCodes
+
+
+def check_memory_layout_row_major(data):
+    if not data.flags['C_CONTIGUOUS']:
+        raise BrainFlowError('wrong memory layout, should be row major, make sure you didnt tranpose array',
+                             BrainflowExitCodes.INVALID_ARGUMENTS_ERROR.value)
```
Comment (xloem) [inline python-package/brainflow/utils.py:10]: so personally I would combine them something like:
```
def check_memory_layout_row_major(data, ndim):
    if not data.flags['C_CONTIGUOUS']:
        raise BrainFlowError('wrong memory layout, should be row major, make sure you didn't transpose array',
                             BrainflowExitCodes.INVALID_ARGUMENTS_ERROR.value)
    if len(data.shape) != ndim:
        raise BrainFlowError('wrong shape for data, should be {}d array'.format(ndim),
                                 BrainflowExitCodes.INVALID_ARGUMENTS_ERROR.value)
```

**Review by xloem — APPROVED** [review]
It's always great to add more checks, assuming deleting that file is correct.

I have a couple medical appointments today and may not address other queued things for a bit.

**Pull request merged**